### PR TITLE
Diet eq

### PR DESCRIPTION
**Pull request by anicolaspp**

### CODE REVIEW

File: core/src/main/scala/cats/collections/Diet.scala [hunk 1]
```diff
@@ -233,10 +234,18 @@ sealed abstract class Diet[A] {
     case DietNode(_, _, r) => r.max
   }
 
-  @deprecated("the semantics of this method are unclear and unspecified, avoid!", "0.8.0")
-  def map[B: Discrete: Order](f: A => B): Diet[B] = this match {
+  //  @deprecated("the semantics of this method are unclear and unspecified, avoid!", "0.8.0")
```
Comment (larsrh) [inline core/src/main/scala/cats/collections/Diet.scala]: I strongly disagree with un-deprecating this. Mapping may destroy the invariants.

File: tests/src/test/scala/cats/collections/DietSpec.scala [hunk 2]
```diff
@@ -101,6 +101,13 @@ class DietSpec extends CatsSuite {
     (Diet.empty[Int] - Range(10, 100)) should be (Diet.EmptyDiet())
   }
 
+  test("diet eq") {
```
Comment (larsrh) [inline tests/src/test/scala/cats/collections/DietSpec.scala]: This demands at least one or two Scalacheck tests.

Comment (anicolaspp) [reply tests/src/test/scala/cats/collections/DietSpec.scala]: I will add some of those

**Comment by larsrh** [comment]
Thanks a lot!

**Comment by anicolaspp** [comment]
I am happy to be back

**Pull request merged**